x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -273,7 +279,8 @@ void getResponse(AsyncSearchId searchId,
 
                 long expirationTime = (long) get.getSource().get(EXPIRATION_TIME_FIELD);
                 String encoded = (String) get.getSource().get(RESULT_FIELD);
-                AsyncSearchResponse response = decodeResponse(encoded, expirationTime);
+                R response = decodeResponse(encoded);
+                response.setExpirationTime(expirationTime);
```

Comment:
funny that this is what Jim had in his original PR and I made him change the constructor to accept an additional expiration time.

Reply:
It makes sense now. I was wondering why expirationTime was made not-final. I think I have a cleaner way of dealing with it. I also fixed a bug here that sneaked in during last update.

Reply:
For the record, I personally find the argument constructor cleaner, but because we need to provide a `Reader` when we pass around the response, it does become cleaner to make the field non-final and set it after creation. What bug do you mean, I may have missed it but you mean it sneaked in as part of this PR?